src/ir/datalog_print_context.h

```diff
@@ -0,0 +1,45 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#ifndef SRC_IR_DATALOG_PRINT_CONTEXT_H_
+#define SRC_IR_DATALOG_PRINT_CONTEXT_H_
+
+#include "absl/strings/str_cat.h"
+
+namespace raksha::ir {
+
+// A class for providing services that require mutable state to be remembered
+// during the Datalog printing process. Currently, is just used for printing
+// unique labels.
+class DatalogPrintContext {
+ public:
+  DatalogPrintContext() : check_counter_(0) {}
+
+  // DatalogPrintContext is neither copyable nor movable.
```

Comment:
It is still movable. I went back and forth on whether to delete the move, as we don't strictly need it right now. I eventually decided to keep the move because disallowing moveability can be very annoying to use in some circumstances and it is unnecessary for protecting the big invariant that we want: that there is only one source of truth on the counter. This is similar to the invariant provided by `unique_ptr`, and that guards memory safety, a much more fundamentally important concern, so if it's allowable there, it's probably fine here too.

Reply:
ah, good call, I'll fix that.

Reply:
Fixed.